e3/electrolyt/plan.py

```diff
@@ -92,14 +99,23 @@ def __init__(self,
         :type host: str | None
         :param target: see e3.env.BaseEnv.set_env
         :type target: str | None
+        :param enabled: whether the plan line is enabled or disabled
+        :type enabled: bool
         :param kwargs: additional data for the current scope/context
         :type kwargs: dict
         """
+        self.ignore_disabled = ignore_disabled
         if stack:
             # This is a scope creation, so copy current env
             self.stack = stack
             self.plan = plan
             new = self.stack[-1].copy(build=build, host=host, target=target)
+
+            if new.enabled and not enabled:
```

Comment:
IIUC, the expected way to use this is to initially create a context with stack=None, and then it will recursively take care of  creating the stack elements; as a result, the BaseEnv objets in the stack should always have an "enabled" attribute. But someone calling this initialization method with a stack he created himself for whatever reason would get an exception, here, because of this attribute being missing. Maybe the easiest is just to document the fact that the BaseEnv object has an additional attribute?

Reply:
We already document that stack is "internal". I'm improving the doc to explicitly say that user instantiation of PlanContext should be done with stack set to None.